packages/zk-eddsa-event-ticket-pcd/test/ZKEdDSAEventTicketPCD.spec.ts

```diff
@@ -0,0 +1,394 @@
+    expect(snarkInputForValidEventIds(undefined)).deep.equal(
+      Array(VALID_EVENT_IDS_MAX_LEN).fill(BABY_JUB_NEGATIVE_ONE.toString())
+    );
+  });
+  it("should handle empty input", async function () {
+    expect(snarkInputForValidEventIds([])).deep.equal(
+      Array(VALID_EVENT_IDS_MAX_LEN).fill(BABY_JUB_NEGATIVE_ONE.toString())
+    );
+  });
+  it("should handle short input", async function () {
+    const expected = Array(VALID_EVENT_IDS_MAX_LEN).fill(
+      BABY_JUB_NEGATIVE_ONE.toString()
+    );
+    const input = [uuid().toString(), uuid().toString(), uuid().toString()];
+    for (let i = 0; i < input.length; i++) {
+      expected[i] = uuidToBigInt(input[i]).toString();
+    }
+    expect(snarkInputForValidEventIds(input)).deep.equal(expected);
+  });
+  it("should handle full input", async function () {
+    const input: string[] = [];
+    const expected: string[] = [];
+    for (let i = 0; i < VALID_EVENT_IDS_MAX_LEN; i++) {
+      const id = uuid();
+      input.push(id.toString());
+      expected.push(uuidToBigInt(id.toString()).toString());
+    }
+    expect(snarkInputForValidEventIds(input)).deep.equal(expected);
+  });
+  it("should reject input if too large", async function () {
+    const input: string[] = [];
+    const expected: string[] = [];
+    for (let i = 0; i < VALID_EVENT_IDS_MAX_LEN + 1; i++) {
+      const id = uuid();
+      input.push(id.toString());
+      expected.push(uuidToBigInt(id.toString()).toString());
+    }
+    assert.throws(() => {
+      snarkInputForValidEventIds(input);
+    });
+  });
+});
+
+describe("ZKEdDSAEventTicketPCD should work", function () {
+  this.timeout(1000 * 30);
+
+  let pcd1: ZKEdDSAEventTicketPCD;
+
+  const ticketData1: ITicketData = {
+    // the fields below are not signed and are used for display purposes
+
+    attendeeName: "test name",
+    attendeeEmail: "user@test.com",
+    eventName: "event",
+    ticketName: "ticket",
+    checkerEmail: "checker@test.com",
+
+    // the fields below are signed using the server's private eddsa key
+
+    ticketId: "b38501b0-cf66-4416-a4ef-631a9a9b32c4",
+    eventId: "4332f16c-8444-4261-94b9-a0ba8ca917e2",
+    productId: "c94002fb-2c41-480b-842d-fa826fb517e5",
+    timestampConsumed: 0,
+    timestampSigned: 1693028498280,
+    attendeeSemaphoreId: identity1.getCommitment().toString(),
+    isConsumed: false,
+    isRevoked: false
+  };
+
+  const fieldsToReveal1: EdDSATicketFieldsToReveal = {
+    revealEventId: true,
+    revealProductId: true
+  };
+
+  const fieldsToReveal2: EdDSATicketFieldsToReveal = {
+    revealEventId: true,
+    revealProductId: true,
+    revealIsConsumed: true,
+    revealIsRevoked: true
+  };
+
+  // validEventIds1 contains ticketData1.eventId
+  const validEventIds1: string[] = [
+    uuid(),
+    ticketData1.eventId,
+    uuid(),
+    uuid()
+  ];
+
+  // validEventIds2 does not contain ticketData1.eventId
+  const validEventIds2: string[] = [uuid(), uuid(), uuid(), uuid()];
```

Comment:
these tests would be more readable if the names here meant something - eg. `validEventIds2` could be named something like `validEventIdsWithoutEvent1`